src/figuresClasses.ts

```diff
@@ -1,11 +1,89 @@
-export interface Figure {}
+type ShapeType = 'Triangle' | 'Circle' | 'Rectangle';
+type ColorType = 'red' | 'green' | 'blue';
 
-export class Triangle implements Figure {}
+export interface Figure {
+  shape: ShapeType;
+  color: ColorType;
 
-export class Circle implements Figure {}
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  public shape: ShapeType = 'Triangle';
+
+  constructor(
+    public color: ColorType,
+    public sideA: number,
+    public sideB: number,
+    public sideC: number,
+  ) {
+    if (sideA <= 0 || sideB <= 0 || sideC <= 0) {
+      throw new Error('All sides must be greater than zero');
+    }
```

Comment:
you can create function like

```
function checkOnError(condition: boolean, errorMessage: string): void {
  if (condition) {
    throw new Error(errorMessage);
  }
}
```
It is needed because you have the same logic in three places.